### PR TITLE
Upgrade to hyper 1

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

File: witchcraft-server/src/body.rs [hunk 1]
```diff
@@ -44,47 +45,56 @@ impl RequestBody {
...
         self: Pin<&mut Self>,
         cx: &mut Context<'_>,
-    ) -> Poll<Result<Option<HeaderMap>, Error>> {
-        self.project()
-            .inner
-            .poll_trailers(cx)
-            .map_err(|e| Error::service_safe(e, ClientIo))
-    }
+    ) -> Poll<Option<Result<Bytes, hyper::Error>>> {
+        let mut this = self.project();
```
Comment (pierceforte) [inline witchcraft-server/src/body.rs:63]: What's the difference between this and the `self.as_mut().project()` below https://github.com/palantir/witchcraft-rust-server/pull/158/files#diff-91aac5a936882d40f7bbd7ae1f10863422f0398383c828a40e5265922f6c08ffR91?

File: witchcraft-server/src/service/gzip.rs [hunk 2]
```diff
@@ -188,57 +188,74 @@ where
...
+                        if this.body.is_end_stream() {
+                            encoder.try_finish().unwrap();
+                            let buf = encoder.get_mut().get_mut().split().freeze();
+                            Poll::Ready(Some(Ok(Frame::data(buf))))
+                        } else {
+                            encoder.flush().unwrap();
+                            let buf = encoder.get_mut().get_mut().split().freeze();
+                            *this.state = State::Compressing(encoder);
+                            Poll::Ready(Some(Ok(Frame::data(buf))))
+                        }
```
Comment (pierceforte) [inline witchcraft-server/src/service/gzip.rs]: nit: maybe do this after the if/else?

```
// assuming you don't need to do this before re-assigning this.sate in the else
let buf = encoder.get_mut().get_mut().split().freeze();
Poll::Ready(Some(Ok(Frame::data(buf))))
```

File: witchcraft-server/src/body.rs (hunk 1 above)
Comment (sfackler) [reply witchcraft-server/src/body.rs:63]: The as_mut allows the `Pin<&mut Self>` to be accessed multiple times. `self.project()` will move out of `self`, while `self.as_mut().project()` won't: https://doc.rust-lang.org/stable/std/pin/struct.Pin.html#method.as_mut

**Review by pierceforte — DISMISSED** [review]

File: witchcraft-server/src/service/gzip.rs (hunk 2 above)
Comment (sfackler) [reply witchcraft-server/src/service/gzip.rs]: We do need to construct the buf before reassigning the state, yeah since it holds the encoder.

**Comment by sfackler** [comment]
We should be good to release once https://github.com/palantir/conjure-rust-runtime/pull/177 lands.

**Comment by sfackler** [comment]
Turns out we need one more tweak to conjure-http: https://github.com/palantir/conjure-rust/pull/356

**Comment by sfackler** [comment]
I think we're good to go!

**Review by pierceforte — APPROVED** [review]
Re-approving

**Pull request merged**